pkg/cmd/roachtest/tests/backup_restore_roundtrip.go

```diff
@@ -372,3 +345,61 @@ func workloadWithCancel(m cluster.Monitor, fn func(ctx context.Context) error) f
 	})
 	return cancelWorkload
 }
+
+// setupClusterForBackupRestore sets up the cluster for backup and restore with
+// some sensible settings. It returns the test utils, the backup restore driver,
+// a function to run the background workload, the databases and tables the
+// workload/restore will work on, and a cleanup function.
+func setupClusterForBackupRestore(
```

Comment:
if a function has a lot of input args and a lot of output args, something is off. what was the motivation for putting this logic into a function? i.e. what was wrong with the initial state?